geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/vector/ArrowAttributeReader.scala

```diff
@@ -174,56 +194,124 @@ object ArrowAttributeReader {
     */
   class ArrowDictionaryIntReader(accessor: NullableIntVector#Accessor,
                                  val dictionary: ArrowDictionary,
-                                 val dictionaryType: TypeBindings) extends ArrowAttributeReader with HasArrowDictionary {
+                                 val dictionaryType: TypeBindings) extends ArrowDictionaryReader {
+    private val holder = new NullableIntHolder
     override def apply(i: Int): AnyRef = {
-      if (accessor.isNull(i)) { null } else {
-        dictionary.lookup(accessor.get(i))
+      accessor.get(i, holder)
+      if (holder.isSet == 0) { null } else {
+        dictionary.lookup(holder.value)
+      }
+    }
+
+    override def getEncoded(i: Int): Int = accessor.get(i)
+  }
+
+  object ArrowGeometryReader {
+    def apply(vector: FieldVector, binding: Class[_], precision: EncodingPrecision): ArrowAttributeReader = {
+      if (binding == classOf[Point]) {
```

Comment:
Idly, I wonder if it'd be 'better' to make ObjectType 'know' about the individual geometry types. 

In the above list, we'd be able to dispatch directly to the Arrow{Point,LineString}Readers, no?